hikari/api/rest.py

```diff
@@ -1094,21 +1265,96 @@ async def fetch_webhook(
         *,
         token: undefined.UndefinedOr[str] = undefined.UNDEFINED,
     ) -> webhooks.Webhook:
-        ...
+        """Fetch an existing webhook.
+
+        Parameters
+        ----------
+        webhook : hikari.snowflakes.SnowflakeishOr[hikari.webhooks.Webhook]
+            The webhook to fetch. This may be a `hikari.webhooks.Webhook`
+            or the ID of an existing webhook.
+        token : hikari.undefined.UndefinedOr[builtins.str]
+            If specified, the webhoook token that will be used to fetch
+            the webhook instead of the stored token.
+
+        Returns
+        -------
+        hikari.webhooks.Webhook
+            The requested webhook.
+
+        Raises
+        ------
+        hikari.errors.Forbidden
+            If you are missing the `MANAGE_WEBHOOKS` permission when not
+            using a token.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the webhook is not found.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_channel_webhooks(
         self,
         channel: snowflakes.SnowflakeishOr[channels.TextChannel],
     ) -> typing.Sequence[webhooks.Webhook]:
-        ...
+        """Fetch all channel webhooks.
+
+        Parameters
+        ----------
+        channel : hikari.snowflakes.SnowflakeishOr[hikari.channels.TextChannel]
+            The channel to fetch the webhooks for. This
+            may be a `hikari.channels.TextChannel` or the ID of an
+            existing channel.
+
+        Returns
+        -------
+        typing.Sequence[hikari.webhooks.Webhook]
+            The fetched webhooks.
+
+        Raises
+        ------
+        hikari.errors.Forbidden
+            If you are missing the `MANAGE_WEBHOOKS` permission.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the channel is not found.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_guild_webhooks(
         self,
         guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
     ) -> typing.Sequence[webhooks.Webhook]:
-        ...
+        """Fetch all guild webhooks.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            The guild to fetch the webhooks for This
+            may be a `hikari.channels.PartialGuild` or the ID of an
```

Comment:
could be more explicit about the contravariance here (and in all other places where you reference specific classes where we expect to get subclasses passed)